generic-file-system/api/src/main/java/org/pentaho/platform/api/genericfile/GetTreeOptions.java

```diff
@@ -247,10 +254,10 @@ public void setExpandedPath( @Nullable String expandedPath ) throws InvalidPathE
    * Moreover, if the {@link #getExpandedMaxDepth() effective expanded path maximum depth} not {@code null}, the result
    * will include a subtree of that maximum depth, rooted at the expanded path.
    *
-   * @param expandedPath The expanded path.
+   * @param expandedPaths A list of expanded paths.
    */
-  public void setExpandedPath( @Nullable GenericFilePath expandedPath ) {
-    this.expandedPath = expandedPath;
+  public void setExpandedPaths( @Nullable List<GenericFilePath> expandedPaths ) {
+    this.expandedPaths = expandedPaths != null && !expandedPaths.isEmpty() ? List.copyOf( expandedPaths ) : null;
```

Comment:
Same as above

Reply:
Here, the defensive programming argument is less defensible, given the list is explicitly passed. Still, that was the spirit.

Reply:
Removed copy in https://github.com/pentaho/pentaho-scheduler-plugin/pull/243/commits/75b3bd168c4084e78908e5ac4431c576bed56446.